lib/checkother.cpp

```diff
@@ -344,24 +344,26 @@ void CheckOther::invalidPointerCast()
             if (fromType->type != toType->type && fromType->type >= ValueType::Type::BOOL && toType->type >= ValueType::Type::BOOL && (toType->type != ValueType::Type::CHAR || printInconclusive)) {
                 if (toType->isIntegral() && fromType->isIntegral())
                     continue;
-                std::string toStr = toType->isIntegral() ? "integer *" : toType->str();
+                std::string toStr = toType->str();
                 toStr.erase(toStr.size()-2);
-                std::string fromStr = fromType->isIntegral() ? "integer *" : fromType->str();
+                if (toStr.rfind("signed ", 0) == 0)
+                    toStr.erase(0, 7);
+                std::string fromStr = fromType->str();
                 fromStr.erase(fromStr.size() - 2);
```

Comment:
The only thing this does it removing the ` *` at the end of the type, and when we print it, we add back `*` (note, without a space). I was unsure if it is worth keeping this or just print `fromType->str()` directly. (which would result in `unsigned int *` (with a space beteween `int` and `*`).

Reply:
ok sounds better to use the builtin `fromType->str()` then. That will give a consistent output for checks.. if we don't want to have redundant space we can remove it there.

Reply:
Done, it also becomes much nicer to avoid the string manipulation in the checker.